Sources/SwiftSyntaxBuilder/Syntax+StringInterpolation.swift

```diff
@@ -160,3 +242,197 @@ extension SyntaxExpressibleByStringInterpolation {
+extension Int8: ExpressibleByLiteralSyntax {}
+extension Int16: ExpressibleByLiteralSyntax {}
+extension Int32: ExpressibleByLiteralSyntax {}
+extension Int64: ExpressibleByLiteralSyntax {}
+extension UInt: ExpressibleByLiteralSyntax {}
+extension UInt8: ExpressibleByLiteralSyntax {}
+extension UInt16: ExpressibleByLiteralSyntax {}
+extension UInt32: ExpressibleByLiteralSyntax {}
+extension UInt64: ExpressibleByLiteralSyntax {}
+
+extension ExpressibleByLiteralSyntax where Self: FloatingPoint, Self: LosslessStringConvertible {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    switch floatingPointClass {
+    case .positiveInfinity:
+      return MemberAccessExpr(name: "infinity")
+
+    case .quietNaN:
+      return MemberAccessExpr(name: "nan")
+
+    case .signalingNaN:
+      return MemberAccessExpr(name: "signalingNaN")
+
+    case .negativeInfinity, .negativeZero:
+      return PrefixOperatorExpr(
+        operatorToken: "-",
+        postfixExpression: (-self).makeLiteralSyntax()
+      )
+
+    case .negativeNormal, .negativeSubnormal, .positiveZero, .positiveSubnormal, .positiveNormal:
+      // TODO: Thousands separators?
+      let digits = String(self)
+      return FloatLiteralExpr(floatingDigits: digits)
+    }
+
+  }
+}
+extension Float: ExpressibleByLiteralSyntax {}
+extension Double: ExpressibleByLiteralSyntax {}
+
+#if !((os(macOS) || targetEnvironment(macCatalyst)) && arch(x86_64))
+@available(macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension Float16: ExpressibleByLiteralSyntax {}
+#endif
+
+extension Bool: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    BooleanLiteralExpr(self)
+  }
+}
+
+extension ArraySlice: ExpressibleByLiteralSyntax where Element: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    ArrayExpr(
+      leftSquare: .leftSquareBracket,
+      elements: ArrayElementList {
+        for elem in self {
+          ArrayElement(expression: elem.makeLiteralSyntax())
+        }
+      },
+      rightSquare: .rightSquareBracket
+    )
+  }
+}
+
+extension Array: ExpressibleByLiteralSyntax where Element: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    self[...].makeLiteralSyntax()
+  }
+}
+
+extension Set: ExpressibleByLiteralSyntax where Element: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    // Sets are unordered. Sort the elements by their source-code representation to emit them in a stable order.
+    let elemSyntaxes = map {
+      $0.makeLiteralSyntax()
+    }.sorted {
+      $0.syntaxTextBytes.lexicographicallyPrecedes($1.syntaxTextBytes)
+    }
+
+    return ArrayExpr(
+      leftSquare: .leftSquareBracket,
+      elements: ArrayElementList {
+        for elemSyntax in elemSyntaxes {
+          ArrayElement(expression: elemSyntax)
+        }
+      },
+      rightSquare: .rightSquareBracket
+    )
```

Comment:
Couldn’t this just be `Array(elemSyntaxes).makeLiteralSyntax()`?

(similar below for `Dictionary` and `KeyValuePairs`)

Reply:
Oh, of course. Makes sense 👍